File: generate_pdf.py
import json
from reportlab.lib.pagesizes import A4
from reportlab.pdfgen import canvas
from reportlab.lib.units import inch
from reportlab.pdfbase import pdfmetrics
from reportlab.pdfbase.ttfonts import TTFont
import sys

def create_pdf(json_file, output_pdf):
    with open(json_file, 'r', encoding='utf-8') as f:
        accounts = json.load(f)

    c = canvas.Canvas(output_pdf, pagesize=A4)
    width, height = A4
    y = height - 50

    c.setFont("Helvetica-Bold", 18)
    c.drawString(50, y, "V-SCENE User Accounts")
    y -= 40

    c.setFont("Helvetica-Bold", 14)
    c.drawString(50, y, "Admin Account")
    y -= 20
    
    c.setFont("Helvetica", 12)
    c.drawString(50, y, "Username: admin")
    y -= 15
    c.drawString(50, y, "Password: password123")
    y -= 30

    c.setFont("Helvetica-Bold", 14)
    c.drawString(50, y, "Student Accounts (VS001 - VS100)")
    y -= 20

    # Draw Table Header
    c.setFont("Helvetica-Bold", 12)
    c.drawString(50, y, "No.")
    c.drawString(100, y, "Username")
    c.drawString(200, y, "Password")
    y -= 20

    c.setFont("Helvetica", 12)
    students = [a for a in accounts if a.get('role') == 'student']
    
    for i, a in enumerate(students):
        if y < 50:
            c.showPage()
            y = height - 50
            c.setFont("Helvetica-Bold", 12)
            c.drawString(50, y, "No.")
            c.drawString(100, y, "Username")
            c.drawString(200, y, "Password")
            y -= 20
            c.setFont("Helvetica", 12)
            
        c.drawString(50, y, str(i + 1))
        c.drawString(100, y, a['username'])
        c.drawString(200, y, a['password'])
        y -= 15

    c.save()

if __name__ == '__main__':
    create_pdf('src/data/accounts.json', '/Users/phakansitketwiset/.gemini/antigravity/brain/04775cac-96ed-45da-a662-c7850d1a0ca8/vscene_accounts.pdf')

File: generate_pdf.js
import fs from 'fs';
const accounts = JSON.parse(fs.readFileSync('src/data/accounts.json', 'utf8'));

let html = `
<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <title>V-SCENE User Accounts</title>
  <style>
    body { font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif; padding: 40px; }
    h1 { color: #2563eb; }
    h2 { color: #334155; border-bottom: 2px solid #e2e8f0; padding-bottom: 8px; }
    table { width: 100%; border-collapse: collapse; margin-top: 20px; }
    th, td { border: 1px solid #cbd5e1; padding: 12px; text-align: left; }
    th { background-color: #f1f5f9; color: #334155; font-weight: bold; }
    tr:nth-child(even) { background-color: #f8fafc; }
    .admin-card { background: #fef2f2; border: 1px solid #fca5a5; padding: 20px; border-radius: 8px; margin-bottom: 30px; }
  </style>
</head>
<body>
  <h1>V-SCENE User Accounts</h1>
  
  <div class="admin-card">
    <h2>👨‍🏫 Admin Account</h2>
    <p><strong>Username:</strong> admin</p>
    <p><strong>Password:</strong> password123</p>
  </div>

  <h2>👨‍⚕️ Student Accounts (VS001 - VS100)</h2>
  <table>
    <thead>
      <tr>
        <th>No.</th>
        <th>Username</th>
        <th>Password</th>
      </tr>
    </thead>
    <tbody>
`;

accounts.filter(a => a.role === 'student').forEach((a, idx) => {
  html += `
      <tr>
        <td>${idx + 1}</td>
        <td><strong>${a.username}</strong></td>
        <td>${a.password}</td>
      </tr>
  `;
});

html += `
    </tbody>
  </table>
</body>
</html>
`;

fs.writeFileSync('accounts_list.html', html);
